### ISSUE TITLE
Sorry, you cannot set a Cypress namespaced cookie.

### ISSUE THREAD

**Issue by amazza64labs** [cypress-io/cypress#2659]
### Current behavior:

```
Error: Sorry, you cannot set a Cypress namespaced cookie.\\n\\nCommon situations why this would fail:\\n  - you don\'t have internet access\\n  - you forgot to run / boot your web server\\n  - your web server isn\'t accessible\\n  - you have weird network configuration settings on your computer\\n\\nThe stack trace for this error is:\\n\\nError: Sorry, you cannot set a Cypress namespaced cookie.\\n    at Object.setCookie (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/cookies.js:72:17)\\n    at /Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/index.js:98:30\\n    at tryCatcher (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)\\n    at Function.Promise.attempt.Promise.try (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/method.js:39:29)\\n    at normalize (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/index.js:89:30)\\n    at /Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/lib/automation/index.js:139:20\\n    at tryCatcher (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)\\n    at Promise._settlePromiseFromHandler (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:510:31)\\n    at Promise._settlePromise (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:567:18)\\n    at Promise._settlePromiseCtx (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:10)\\n    at Async._drainQueue (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:138:12)\\n    at Async._drainQueues (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:143:10)\\n    at Immediate.Async.drainQueues (/Users/current_user/Library/Caches/Cypress/3.1.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/bluebird/js/release/async.js:17:14)\\n    at runCallback (timers.js:781:20)\\n    at tryOnImmediate (timers.js:743:5)\\n    at processImmediate [as _immediateCallback] (timers.js:714:5)\\n",\n      "date_recorded": "2018-10-25T15:45:55-04:00",\n      
```

This is during the `cy.visit('https://www.carparts.com');`

### Desired behavior:

I would like cypress to execute the action "visit" with `https://www.carparts.com`

### Steps to reproduce:

1) Create a simple script

```js
it('test', () => {
   cy.visit(`https://www.carparts.com`);
});
```

2) Run the script, receive an error

### Versions

3.1.0, Chrome, latest

**Comment by jennifer-shehane** [comment]
We run `cy.clearCookies()` before each test. Within this `cy.clearCookies()`, we throw an error if the cookie `isNamespaced` - which initially checks that the name of the cookie is not `undefined`, then continues to check it it's within the cypress namespace. 

It's finding the cookie below (with an `undefined` name) on this site which I believe is making it throw this confusing error. 

```
{ 
  value: 'deleted',
  expires: '1970-01-01T00:00:01.000Z',
  domain: 'www.carparts.com',
  path: '/',
  hostOnly: true,
  creation: '2018-11-01T19:48:02.086Z',
  lastAccessed: '2018-11-01T19:48:02.086Z',
  url: 'https://www.carparts.com/',
  name: undefined,
  expiry: 1 
}
```

You can also see this in the cookies when visiting the site:

![screen shot 2018-11-01 at 3 49 10 pm](https://user-images.githubusercontent.com/1271364/47875974-5999b280-ddee-11e8-9a60-306afe2846da.png)

**Comment by jennifer-shehane** [comment]
I think this line of code should say `return cookie if not name = cookie?.name`, because if it returns the `cookie`, then that will evaluate as true and always evaluate as `isNamespaced` being true, when really we want this to return `false`

**Comment by jennifer-shehane** [comment]
I've opened a preliminary PR for this fix. Needs a test https://github.com/cypress-io/cypress/pull/2709

**Comment by jennifer-shehane** [comment]
A PR is in to fix this.

Just so you know - here is the cookie with the no name. 

![screen shot 2018-11-01 at 5 20 11 pm](https://user-images.githubusercontent.com/1271364/47880805-9e2b4b00-ddfa-11e8-8cd7-6613d3f4e386.png)

**Comment by jennifer-shehane** [comment]
This was fixed in [3.1.1](https://docs.cypress.io/guides/references/changelog.html#3-1-1)
